src/Psr/Tracing/RelayOpenTelemetry.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CacheWerk\Relay\Psr\Tracing;
+
+use Relay\Relay;
+
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\API\Trace\TracerInterface;
+
+class RelayOpenTelemetry
+{
+    /**
+     * The Relay instance.
+     *
+     * @var \Relay\Relay
+     */
+    protected Relay $relay;
+
+    /**
+     * The OpenTelemetry tracer instance.
+     *
+     * @var \OpenTelemetry\API\Trace\TracerInterface
+     */
+    protected TracerInterface $tracer;
+
+    /**
+     * Creates a new `NewRelicRelay` instance.
+     *
+     * @param  \Relay\Relay  $relay
+     * @param  \OpenTelemetry\API\Trace\TracerInterface  $tracer
+     * @return void
+     */
+    public function __construct(Relay $relay, TracerInterface $tracer)
+    {
+        $this->relay = new $relay;
+        $this->tracer = new $tracer;
+    }
+
+    /**
+     * Executes Relay methods inside OpenTelemetry span.
+     *
+     * @param  string  $name
+     * @param  array  $arguments
+     * @return mixed
+     */
+    public function __call(string $name, array $arguments)
+    {
+        $span = $this->tracer->spanBuilder('Relay') // TODO: What's the name?
+            ->setAttribute('command', strtolower($name))
+            ->setSpanKind(SpanKind::KIND_CLIENT) // TODO: Is this the correct kind?
+            ->startSpan()
+            ->activate(); // TODO: Is this needed?
```

Comment:
Basically starting/ending the span is just starting/ending the span timer to measure span duration. Activating the span is marking it as active in some global span context storage, for example:

```php
$main = $tracer->spanBuilder('main-operation')->startSpan();
var_dump(get_class(Span::getCurrent())); // NonRecordingSpan empty span

$main->activate();
var_dump(get_class(Span::getCurrent())); // OpenTelemetry\SDK\Trace\Span our span
```

The active or the current span will later be used as a parent span for other spans created while the span is active. Only one span can be active and if there none, then the new root span is created.

I guess they 2 have 2 separate operations to activate and starta a span, because there is asynchronous PHP where it makes sense. I don't know much about it, but Otel has this [example](https://github.com/open-telemetry/opentelemetry-php/blob/be3f88fa6cb6cb2f76e1e6da520eac30535e6488/tests/Integration/Context/Fiber/test_context_switching_ffi_observer.phpt).

Reply:
Leaving it out for now.